### PR TITLE
feat/inbox enhancement

### LINKED ISSUE #278: inbox ui enhancement
- [ ] the icons move by the title ( dynamic)
- [ ] double click menu ( move, plan, delete, expand)
- [ ] cicle mark as done ( item's desappear)

### LINKED ISSUE #280: Inbox: minor functionality enhancements
- [ ] mark as done with icon, on menu
- [ ] delete on menu
- [ ] expand on menu

### PR DESCRIPTION
**Pull request by deewakar-k**

### CODE REVIEW

**Review by immadsahin — CHANGES_REQUESTED** [review]
- [x]     its showing all icons by default, where it should show by default.
- [x] no need to change the move icon
- [x] the small source icon can be hidden now, dont add a placeholder until functional. it looks ungy.
- [x]     the menu is opening in wrong place.
- [x] fix menu for mac user

**Comment by deewakar-k** [comment]
> * [x]      its showing all icons by default, where it should show by default.
> 
>     * [x]  no need to change the move icon
> 
>     * [x]  the small source icon can be hidden now, dont add a placeholder until functional. it looks ungy.
> 
>     * [x]      the menu is opening in wrong place.
> 
>     * [x]  fix menu for mac user

done

**Review by immadsahin — DISMISSED** [review]

File: apps/frontend/src/components/Inbox/InboxItems.tsx [hunk 1]
```diff
@@ -34,11 +41,26 @@ export const InboxItems: React.FC = () => {
...
+    (item: InboxItem) => {
+      setSelectedItem(item)
+    },
+    [setSelectedItem]
+  )
+
+  const handleRightClick = useCallback(
+    (e: React.MouseEvent, item: InboxItem) => {
+      e.preventDefault()
+      setOpenDropdownId(item._id)
```
Comment (madhav-relish) [inline apps/frontend/src/components/Inbox/InboxItems.tsx]: You'll need to add null here in or condition

image

File: apps/frontend/src/components/Inbox/InboxItems.tsx [hunk 2]
```diff
@@ -34,11 +41,26 @@ export const InboxItems: React.FC = () => {
     if (!isFetched) {
       fetchInbox()
     }
-  }, [session, fetchInboxData, setIsFetched])
+  }, [isFetched, fetchInbox])
```
Comment (madhav-relish) [inline apps/frontend/src/components/Inbox/InboxItems.tsx]: Add proper error logging for better debugging like
console.error("Error from fetchInbox::", error)

**Review by madhav-relish — CHANGES_REQUESTED** [review]
Can you check this also, if an item is selected and we click on that again and again, every time an api is being triggered. 


https://github.com/user-attachments/assets/5eb011e4-bf18-4473-b72d-04153eb07fa7

**Comment by madhav-relish** [comment]
@oliursahin  what do you think about this component? https://ui.shadcn.com/docs/components/drawer

It will be great for mobile devices as well right out of the box

**Comment by deewakar-k** [comment]
> Can you check this also, if an item is selected and we click on that again and again, every time an api is being triggered. 
> 
> 
> 
> 
> 
> https://github.com/user-attachments/assets/5eb011e4-bf18-4473-b72d-04153eb07fa7
> 
> 
> 
> 

i believe this is the pr for ui not the integration, integration was done in earlier prs

**Comment by sajdakabir** [comment]
> > Can you check this also, if an item is selected and we click on that again and again, every time an api is being triggered.
> > 
> >   
> >     
> >     
> > 
> >     Screen.Recording.2024-10-12.131732.mp4
> 
> i believe this is the pr for ui not the integration, integration was done in earlier prs


so what? you can fix it, isn't it?!  @deewakar-k

**Review by madhav-relish — DISMISSED** [review]

**Pull request merged**